planning/mission_planner/src/rerouting_static_obstacle/rerouting_static_obstacle.cpp

```diff
@@ -0,0 +1,206 @@
+: Node("rerouting_static_obstacle", node_options)
+{
+  using std::placeholders::_1;
+
+  sub_odom_ = this->create_subscription<nav_msgs::msg::Odometry>(
+    "input/odometry", rclcpp::QoS(1), std::bind(&ReroutingStaticObstacle::on_odom, this, _1));
+
+  sub_route_ = create_subscription<autoware_planning_msgs::msg::LaneletRoute>(
+    "input/route", rclcpp::QoS{1}.transient_local(),
+    std::bind(&ReroutingStaticObstacle::route_callback, this, std::placeholders::_1));
+
+  map_subscriber_ = create_subscription<HADMapBin>(
+    "input/vector_map", rclcpp::QoS{10}.transient_local(),
+    std::bind(&ReroutingStaticObstacle::map_callback, this, std::placeholders::_1));
+
+  sub_trigger_ = this->create_subscription<geometry_msgs::msg::PointStamped>(
+    "input/reroute_point", rclcpp::QoS(1),
+    std::bind(&ReroutingStaticObstacle::on_trigger, this, _1));
+
+  const auto adaptor = component_interface_utils::NodeAdaptor(this);
+  adaptor.init_cli(cli_change_route_);
+}
+
+void ReroutingStaticObstacle::on_odom(const nav_msgs::msg::Odometry::SharedPtr msg)
+{
+  current_pose_ = msg->pose.pose;
+}
+
+void ReroutingStaticObstacle::route_callback(
+  const autoware_planning_msgs::msg::LaneletRoute::ConstSharedPtr msg)
+{
+  goal_pose_ = msg->goal_pose;
+}
+
+void ReroutingStaticObstacle::map_callback(const HADMapBin::ConstSharedPtr msg)
+{
+  route_handler_.setMap(*msg);
+  lanelet_map_ptr_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(
+    *msg, lanelet_map_ptr_, &traffic_rules_ptr_, &routing_graph_ptr_);
+  lanelet::ConstLanelets all_lanelets = lanelet::utils::query::laneletLayer(lanelet_map_ptr_);
+  road_lanelets_ = lanelet::utils::query::roadLanelets(all_lanelets);
+}
+
+void ReroutingStaticObstacle::on_trigger(const geometry_msgs::msg::PointStamped::SharedPtr msg)
+{
+  geometry_msgs::msg::Pose selected_point;
+  selected_point.position = msg->point;
+
+  lanelet::ConstLanelets selected_point_lanelets;
+  bool selected_point_lanelet_found{false};
+  selected_point_lanelet_found =
+    get_selected_point_lanelets(selected_point, selected_point_lanelets);
+
+  lanelet::ConstLanelets remaining_route_lanelets;
+  bool remaining_route_lanelets_found{false};
+  remaining_route_lanelets_found = get_remaining_route_lanelets(remaining_route_lanelets);
+
+  lanelet::ConstLanelet selected_point_lanelet;
+  bool selected_point_in_route{false};
+  if (selected_point_lanelet_found && remaining_route_lanelets_found) {
+    selected_point_in_route = is_selected_point_in_route(
+      selected_point_lanelets, remaining_route_lanelets, selected_point_lanelet);
+  } else {
+    return;
+  }
+
+  if (selected_point_in_route) {
+    lanelet::routing::LaneletPath alternative_route_lanelets;
+    bool alternative_route_found{false};
+    alternative_route_found =
+      search_alternative_route(selected_point_lanelet, alternative_route_lanelets);
```

Comment:
maybe simplify like this?
```suggestion
    auto alternative_route_found =
      search_alternative_route(selected_point_lanelet, alternative_route_lanelets);
```